src/System.Net.Security/src/System/Net/SecureProtocols/SslState.cs

```diff
@@ -804,14 +807,14 @@ private void CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtoc
         {
             if (message.Failed)
             {
-                StartSendAuthResetSignal(null, asyncRequest, new AuthenticationException(SR.net_auth_SSPI, message.GetException()));
+                StartSendAuthResetSignal(null, asyncRequest, ExceptionDispatchInfo.Capture(new AuthenticationException(SR.net_auth_SSPI, message.GetException())));
```

Comment:
Given that this exception hasn't actually been thrown, this doesn't actually capture anything, does it?